From 9d37434b9242e4e545b1d7a72b6d5b3589fb1c4f Mon Sep 17 00:00:00 2001
From: Russ Weight <russell.h.weight@intel.com>
Date: Wed, 5 Oct 2022 10:04:27 -0700
Subject: [PATCH 83/93] fpga: m10bmc-sec: Reference staging size variable

The secure update driver does a sanity-check of the image size in
comparison to the size of the staging area in FLASH. Change the
staging size reference to a variable instead of a constant in order
to more readily support future, larger FPGAs.

Signed-off-by: Russ Weight <russell.h.weight@intel.com>
---
 drivers/fpga/intel-m10-bmc-sec-update.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/fpga/intel-m10-bmc-sec-update.c b/drivers/fpga/intel-m10-bmc-sec-update.c
index 38649b6a5e64..caca5e4f3f33 100644
--- a/drivers/fpga/intel-m10-bmc-sec-update.c
+++ b/drivers/fpga/intel-m10-bmc-sec-update.c
@@ -1391,7 +1391,7 @@ static enum fw_upload_err m10bmc_sec_prepare(struct fw_upload *fwl,
 
 	sec->cancel_request = false;
 
-	if (!size || size > M10BMC_STAGING_SIZE)
+	if (!size || size > staging_size(sec->m10bmc))
 		return FW_UPLOAD_ERR_INVALID_SIZE;
 
 	ret = rsu_check_idle(sec);
-- 
2.39.0

